### PR TITLE
Fix(UI) #8555: Tags removal issue

### PR DESCRIPTION
**Pull request by aniketkatkar97**
### Describe your changes :

I worked on the following:
- [x] Fixed Error, thrown while removing tags from the data asset details page
- [x] Fixed bug where users were not able to remove the tags on the Dashboard Details page
- [x] Changes in MlModelFeaturesList component:
   - Tag removal bug fix
   - Removed tailwind use and replaced with antd
   - Code optimization
   - Localisation changes

#
### Type of change :

- [x] Bug fix
- [x] Improvement

#
### Frontend Preview (Screenshots) :
<p align="center">

Table Details Page:

https://user-images.githubusercontent.com/51777795/204544976-f7efff38-59c8-43c7-8af0-753bdf51d463.mov

Dashboard Details Page:

https://user-images.githubusercontent.com/51777795/204545042-0fe2f87a-c499-470a-ab07-f13e067dd8fb.mov

Ml Model Details Page:

https://user-images.githubusercontent.com/51777795/204545049-b0f57287-f80f-4bef-ad7b-2c0e50339725.mov

SourceList component: 

https://user-images.githubusercontent.com/51777795/204707749-cac0457c-ce65-498c-9693-edb456b15c52.mov

</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

Frontend: @open-metadata/ui

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/DashboardDetails/DashboardDetails.component.tsx [hunk 1]
```diff
@@ -610,39 +610,46 @@ const DashboardDetails = ({
...
-                  });
-                }}
-              />
-            )}
-          </div>
-        ),
+        render: (tags: Dashboard['tags'], record, index) => {
+          tags?.sort((tag1, tag2) =>
+            tag1.tagFQN.toLowerCase() < tag2.tagFQN.toLowerCase() ? -1 : 1
+          );
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/DashboardDetails/DashboardDetails.component.tsx:616]: - write a reuseable comparator function to sort the tags
- Pull out sorting from the render, Sort once and pass it to the table so we won't be ending up sorting again & again

File: openmetadata-ui/src/main/resources/ui/src/components/EntityTable/EntityTable.component.tsx [hunk 2]
```diff
@@ -531,6 +531,10 @@ const EntityTable = ({
 
   const renderTags: TableCellRendered<Column, 'tags'> = useCallback(
     (tags, record: Column, index: number) => {
+      tags?.sort((tag1, tag2) =>
+        tag1.tagFQN.toLowerCase() < tag2.tagFQN.toLowerCase() ? -1 : 1
+      );
+
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/EntityTable/EntityTable.component.tsx:537]: Should be sorted outside
Use common util for sorting

File: openmetadata-ui/src/main/resources/ui/src/components/MlModelDetail/SourceList.component.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+import { Col, Row, Space, Typography } from 'antd';
+import React, { useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Link } from 'react-router-dom';
+import { EntityType } from '../../enums/entity.enum';
+import { MlFeature } from '../../generated/entity/data/mlmodel';
+import { getEntityLink } from '../../utils/TableUtils';
+import './SourceList.style.less';
+
+const SourceList = ({ feature }: { feature: MlFeature }) => {
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/MlModelDetail/SourceList.component.tsx:24]: I am not sure if you did attach a recording for the same, Can you help me to identify it in screenshots or recordings?
P.S.: **Also, DO NOT CLUB MULTIPLE ISSUE ON SINGLE PR & IF YOU DO PLEASE USE RELEVANT PR TITLE**

File: openmetadata-ui/src/main/resources/ui/src/locale/languages/en-us.json [hunk 4]
```diff
@@ -328,6 +328,11 @@
     "use-aws-credentials": "Use Aws Credentials",
     "use-ssl": "Use SSL",
     "verify-certs": "Verify Certs",
+    "source-plural": "Sources",
+    "data-source": "Data Source",
+    "features-used": "Features Used",
+    "edit-tags": "Edit Tags",
+    "edit-feature": "Edit Feature",
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/locale/languages/en-us.json:335]:
```suggestion
    "edit-entity": "Edit {{entity}}",
```

File: openmetadata-ui/src/main/resources/ui/src/styles/app.less [hunk 5]
```diff
@@ -186,6 +200,9 @@
 .h-full {
   height: 100%;
 }
+.h-auto {
+  height: auto;
+}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/styles/app.less:205]: Move it to size.less

**Review by chirag-madlani — CHANGES_REQUESTED** [review]

File: openmetadata-ui/src/main/resources/ui/src/components/MlModelDetail/SourceList.component.tsx (hunk 3 above)
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/components/MlModelDetail/SourceList.component.tsx:24]: I've added a separate recording for the SourceList Component after the MLModel recording. Please do check.

**Comment by aniketkatkar97** [comment]
@chirag-madlani I've pushed the suggested changes. Can you please review it?

File: openmetadata-ui/src/main/resources/ui/src/components/DashboardDetails/DashboardDetails.component.tsx [hunk 6]
```diff
@@ -586,7 +586,7 @@ const DashboardDetails = ({
               <Tooltip
                 title={
                   dashboardPermissions.EditAll
-                    ? t('label.edit-description')
+                    ? t('label.edit-entity', { entity: 'Description' })
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/DashboardDetails/DashboardDetails.component.tsx:589]:
```suggestion
                    ? t('label.edit-entity', { entity: t('label.description')})
```

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx [hunk 7]
```diff
@@ -251,7 +251,7 @@ const CustomControls: FC<ControlProps> = ({
               disabled={!hasEditAccess}
               title={
                 hasEditAccess
-                  ? t('label.edit-lineage')
+                  ? t('label.edit-entity', { entity: 'Lineage' })
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx:254]:
```suggestion
                  ? t('label.edit-entity', { entity: t('label.lineage') })
```

File: openmetadata-ui/src/main/resources/ui/src/components/EntityTable/EntityTable.component.tsx [hunk 8]
```diff
@@ -702,7 +702,9 @@ const EntityTable = ({
       />
       {editColumn && (
         <ModalWithMarkdownEditor
-          header={`${t('label:edit-column')}: "${editColumn.column.name}"`}
+          header={`${t('label.edit-entity', { entity: 'Column' })}: "${
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/EntityTable/EntityTable.component.tsx:705]:
```suggestion
          header={`${t('label.edit-entity', { entity: t('label.column') })}: "${
```

File: openmetadata-ui/src/main/resources/ui/src/components/MlModelDetail/MlModelFeaturesList.tsx [hunk 9]
```diff
@@ -163,220 +174,201 @@ const MlModelFeaturesList: FC<MlModelFeaturesListProp> = ({
...
+                                    title="Edit"
+                                    width="16px"
+                                  />
+                                </Button>
+                              </Tooltip>
+                            ) : (
+                              <Tooltip
+                                title={
+                                  permissions.EditAll || permissions.EditTags
+                                    ? t('label.edit-entity', { entity: 'Tags' })
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/MlModelDetail/MlModelFeaturesList.tsx:275]:
```suggestion
                                    ? t('label.edit-entity', { entity: t('label.tags') })
```

File: openmetadata-ui/src/main/resources/ui/src/components/MlModelDetail/MlModelFeaturesList.tsx [hunk 10]
```diff
@@ -163,220 +174,201 @@ const MlModelFeaturesList: FC<MlModelFeaturesListProp> = ({
...
+                                  disabled={
+                                    !(
+                                      permissions.EditAll ||
+                                      permissions.EditTags
+                                    )
+                                  }
+                                  type="text">
+                                  <Tags
+                                    startWith="+ "
+                                    tag="Add tag"
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/MlModelDetail/MlModelFeaturesList.tsx:289]: localisation

File: openmetadata-ui/src/main/resources/ui/src/components/MlModelDetail/MlModelFeaturesList.tsx [hunk 11]
```diff
@@ -163,220 +174,201 @@ const MlModelFeaturesList: FC<MlModelFeaturesListProp> = ({
...
+                    <SourceList feature={feature} />
+                  </Col>
+                </Row>
+              </Card>
+            </Col>
+          ))}
+        </Row>
+        {!isEmpty(selectedFeature) && editDescription && (
+          <ModalWithMarkdownEditor
+            header={`${t('label.edit-entity', { entity: 'Feature' })}: "${
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/MlModelDetail/MlModelFeaturesList.tsx:354]:
```suggestion
            header={`${t('label.edit-entity', { entity: t('label.feature') })}: "${
```

File: openmetadata-ui/src/main/resources/ui/src/components/PipelineDetails/PipelineDetails.component.tsx [hunk 12]
```diff
@@ -1012,7 +1012,7 @@ const PipelineDetails = ({
 
       {editTask && (
         <ModalWithMarkdownEditor
-          header={`${t('label.edit-task')}: "${
+          header={`${t('label.edit-entity', { entity: 'Task' })}: "${
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/PipelineDetails/PipelineDetails.component.tsx:1015]: same as suggested above

File: openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamDetailsV1.tsx [hunk 13]
```diff
@@ -946,7 +946,7 @@ const TeamDetailsV1 = ({
                   title={
                     entityPermissions.EditAll ||
                     entityPermissions.EditDisplayName
-                      ? t('label.edit-display-name')
+                      ? t('label.edit-entity', { entity: 'Display Name' })
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamDetailsV1.tsx:949]: same as suggested above

File: openmetadata-ui/src/main/resources/ui/src/pages/KPIPage/EditKPIPage.tsx [hunk 14]
```diff
@@ -427,7 +427,7 @@ const AddKPIPage = () => {
           </Col>
           <Col className="tw-mt-4" span={4}>
             <Typography.Paragraph className="tw-text-base tw-font-medium">
-              {t('label.edit-kpi')}
+              {t('label.edit-entity', { entity: 'KPI' })}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/KPIPage/EditKPIPage.tsx:430]: same as suggested above

File: openmetadata-ui/src/main/resources/ui/src/pages/service/index.tsx [hunk 15]
```diff
@@ -1111,7 +1111,7 @@ const ServicePage: FunctionComponent = () => {
                         <Tooltip
                           title={
                             servicePermission.EditAll
-                              ? t('label.edit-connection')
+                              ? t('label.edit-entity', { entity: 'Connection' })
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/service/index.tsx:1114]: same as suggested above

File: openmetadata-ui/src/main/resources/ui/src/pages/service/index.tsx [hunk 16]
```diff
@@ -1120,7 +1120,7 @@ const ServicePage: FunctionComponent = () => {
                             disabled={!servicePermission.EditAll}
                             type="primary"
                             onClick={handleEditConnection}>
-                            {t('label.edit-connection')}
+                            {t('label.edit-entity', { entity: 'Connection' })}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/service/index.tsx:1123]: same as suggested above

File: openmetadata-ui/src/main/resources/ui/src/styles/components/size.less [hunk 17]
```diff
@@ -11,9 +11,109 @@
...
+}
+.h-7 {
+  height: 28px;
+}
+.h-8 {
+  height: 32px;
+}
+.h-9 {
+  height: 36px;
+}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/styles/components/size.less:93]: Thanks for moving it all

File: openmetadata-ui/src/main/resources/ui/src/utils/CommonUtils.tsx [hunk 18]
```diff
@@ -1055,3 +1055,9 @@ export const getTrimmedContent = (content: string, limit: number) => {
 
   return refinedContent.join(' ');
 };
+
+export const sortTagsCaseInsensitive = (tags: TagLabel[]) => {
+  return tags.sort((tag1, tag2) =>
+    tag1.tagFQN.toLowerCase() < tag2.tagFQN.toLowerCase() ? -1 : 1
+  );
+};
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/utils/CommonUtils.tsx:1063]: add unit tests for the new functions

File: openmetadata-ui/src/main/resources/ui/src/utils/DashboardDetailsUtils.ts [hunk 19]
```diff
@@ -64,3 +66,10 @@ export const getCurrentDashboardTab = (tab: string) => {
 
   return currentTab;
 };
+
+export const sortTagsForCharts = (charts: ChartType[]) => {
+  return charts.map((chart) => ({
+    ...chart,
+    tags: sortTagsCaseInsensitive(chart.tags || []),
+  }));
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/utils/DashboardDetailsUtils.ts:74]: unit tests

**Review by chirag-madlani — CHANGES_REQUESTED** [review]

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**